src/common/templates/_eso-secrets-helper.tpl

```diff
@@ -239,3 +239,141 @@ spec:
     {{- end }}
 {{- end }}
 {{- end }}
+
+{{/*
+Generates Random secrets for the provided secret keys in helpers.tpl or direct usage in secret file
+USAGE:
+{{- include "harnesscommon.generateSecretsInHelpers" (dict "ctx" $) }}
+*/}}
+{{- define "harnesscommon.generateSecretsInHelpers" }}
+{{- $ := .ctx }}
+{{- $hasAtleastOneSecret := false }}
+{{- $localESOSecretCtxIdentifier := (include "harnesscommon.secrets.localESOSecretCtxIdentifier" (dict "ctx" $ )) }}
+{{- range $key, $value := .ctx.Values.secrets.default }}
+    {{- $hasAtleastOneSecret = true }}
+    {{- $pathTosSecret := printf "secrets.default.%s" $key }}
+{{ $key }}: {{ include "harnesscommon.secrets.passwords.manage" (dict "secret" $.Chart.Name "key" $key "providedValues" (list $pathTosSecret ) "length" 10 "context" $) }}
+{{- end }}
+{{- end }}
+
+{{/*
+Renders secrets as environment variables for the provided secret keys in deployment.yaml/sts.yaml
+USAGE:
+{{- include "harnesscommon.renderSecretsAsEnvironmentVariables" (dict "ctx" $) }}
+*/}}
+{{- define "harnesscommon.renderSecretsAsEnvironmentVariables" }}
+{{- $ := .ctx }}
+{{- $hasAtleastOneSecret := false }}
+{{- $localESOSecretCtxIdentifier := (include "harnesscommon.secrets.localESOSecretCtxIdentifier" (dict "ctx" $ )) }}
+{{- $defaultSecretList := .ctx.Values.secrets.default }}
+{{- $kubernetesSecretsList := (index .ctx.Values.secrets.kubernetesSecrets 0).keys }}
```

Comment:
can there be a usecase where there are multiple elements under `kubernetesSecrets` or `externalSecretsOperator` ?

Reply:
yes, it is currently an issue for both eso and kubernetes secret. it requires a huge chunk of repeated code. Will discuss offline on how we can solve it?